src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractServiceImpl.java

```diff
@@ -98,21 +122,142 @@ protected void logDetailedInfoForConcurStandardAccountingExtractFile(ConcurStand
-        return true;
+    public String extractPdpFeedFromStandardAccountingExtract(ConcurStandardAccountingExtractFile concurStandardAccountingExtractFile) {
+        boolean success = true;
+        String pdpFileName = StringUtils.EMPTY;
+        if (!concurStandardAccountingExtractFile.getConcurStandardAccountingExtractDetailLines().isEmpty()){
+            PdpFeedFileBaseEntry pdpFeedFileBaseEntry = buildPdpFeedFileBaseEntry(concurStandardAccountingExtractFile);
+            pdpFileName = buildPdpOutputFileName(concurStandardAccountingExtractFile.getOriginalFileName());
+            String pdpFilePath = getPaymentImportDirectory() + pdpFileName;
+            success = marshalPdpFeedFile(pdpFeedFileBaseEntry, pdpFilePath);
+        }
+        return success ? pdpFileName : StringUtils.EMPTY;
+    }
+    
+    private PdpFeedFileBaseEntry buildPdpFeedFileBaseEntry(ConcurStandardAccountingExtractFile concurStandardAccountingExtractFile) {
+        KualiDecimal pdpTotal = KualiDecimal.ZERO;
+        PdpFeedFileBaseEntry pdpFeedFileBaseEntry = new PdpFeedFileBaseEntry();
+        pdpFeedFileBaseEntry.setVersion(ConcurConstants.StandardAccountingExtractPdpConstants.FEED_FILE_ENTRY_HEADER_VERSION);
+        pdpFeedFileBaseEntry.setHeader(
+                getConcurStandardAccountExtractPdpEntryService().buildPdpFeedHeaderEntry(concurStandardAccountingExtractFile.getBatchDate()));
+        for (ConcurStandardAccountingExtractDetailLine line : concurStandardAccountingExtractFile.getConcurStandardAccountingExtractDetailLines()) {
+            replacePendingClientWithOverride(line);
+            if (StringUtils.equalsIgnoreCase(line.getPaymentCode(), ConcurConstants.PAYMENT_CODE_CASH)) {
+                if (getConcurStandardAccountingExtractValidationService().validateConcurStandardAccountingExtractDetailLine(line)) {
+                    buildAndUpdateAccountingEntryFromLine(pdpFeedFileBaseEntry, line);
+                    pdpTotal = pdpTotal.add(line.getJournalAmount());
+                }
+            }
+        }
+        pdpFeedFileBaseEntry.setTrailer(getConcurStandardAccountExtractPdpEntryService().buildPdpFeedTrailerEntry(pdpFeedFileBaseEntry, pdpTotal));
+        return pdpFeedFileBaseEntry;
+    }
+    
+    private void replacePendingClientWithOverride(ConcurStandardAccountingExtractDetailLine line) {
+        if (StringUtils.equalsIgnoreCase(line.getJournalAccountCode(), ConcurConstants.PENDING_CLIENT)) {
+            String overrideObjectCode = getParameterService().getParameterValueAsString(CUKFSConstants.ParameterNamespaces.CONCUR, 
+                    CUKFSParameterKeyConstants.ALL_COMPONENTS, ConcurConstants.ParameterNames.CONCUR_PENDING_CLIENT_OBJECT_CODE_OVERRIDE);
+            LOG.error("replacePendingClientWithOverride, found a Pending Client object code, replacing it with " + overrideObjectCode);
+            line.setJournalAccountCode(overrideObjectCode);
```

Comment:
The SAE-to-Collector processing also needs to know which lines have the "Pending Client" object code, so setting the replacement on the detail POJO directly will prevent the Collector side from reusing that POJO unless you're doing object-copying somewhere. Also, you should confirm with Sherry whether such lines need to stand apart from others for PDP (since on the Collector side I need such lines to remain standalone in the final file).

Depending on what changes are needed for this processing, the validation code may also need to be updated to handle "Pending Client" object codes.